pulsar-client/src/main/java/org/apache/pulsar/client/impl/RoundRobinPartitionMessageRouterImpl.java

```diff
@@ -34,25 +30,18 @@
             AtomicIntegerFieldUpdater.newUpdater(RoundRobinPartitionMessageRouterImpl.class, "partitionIndex");
     private volatile int partitionIndex = 0;
 
-    public RoundRobinPartitionMessageRouterImpl(boolean useMurmurHash) {
-        super(useMurmurHash);
+    public RoundRobinPartitionMessageRouterImpl(ProducerConfiguration.HashingScheme hashingScheme) {
+        super(hashingScheme);
         PARTITION_INDEX_UPDATER.set(this, 0);
     }
 
     @Override
     public int choosePartition(Message msg, TopicMetadata topicMetadata) {
         // If the message has a key, it supersedes the round robin routing policy
         if (msg.hasKey()) {
-            if (useMurmurHash) {
-                HashFunction hf = Hashing.murmur3_32(0);
-                HashCode hc = hf.hashString(msg.getKey(), StandardCharsets.UTF_8);
-                long l = Integer.toUnsignedLong(hc.asInt());
-                return (int) (l % topicMetadata.numPartitions());
-            }
-            else {
-                return ((msg.getKey().hashCode() & Integer.MAX_VALUE) % topicMetadata.numPartitions());
-            }
+            return (int) (hash.makeHash(msg.getKey()) % topicMetadata.numPartitions());
```

Comment:
This depends on the `makeHash()` to always return a positive number, which is true in the current implementation. We should mention that in the `makeHash()` javadoc.